doc/doxygen/src/Configuration.dox

```diff
@@ -0,0 +1,174 @@
+ *
+ * This page covers how to configure OpenAssetIO for a specific
+ * end-user working environment.
+ *
+ * @section runtime_configuration_overview Overview
+ *
+ * @startuml
+ * node Host {
+ *   package OpenAssetIO {
+ *     collections ManagerPlugin
+ *   }
+ * }
+ *
+ * cloud "Asset Management System" as ams
+ *
+ * ManagerPlugin -right-> ams
+ * ams -left-> ManagerPlugin
+ * @enduml
+ *
+ * An OpenAssetIO enabled tool or application is known as a @ref host "Host"
+ * (as it "hosts" the API, tenuous - but naming things is hard). Through
+ * the use of a plugin system, OpenAssetIO allows the host to
+ * communicate with one or more @ref asset_management_system "Asset Management Systems",
+ * called @ref manager "Managers".
+ *
+ * This page covers hows to configure OpenAssetIO so that the hosts you
+ * use (e.g. @ref DCC "DCC tools", applications and other scripts) use a
+ * specific manager of your choice to @ref resolve and potentially @ref
+ * publish data.
+ *
+ * OpenAssetIO itself is usually built into any host that supports it,
+ * this means you don't normally need do to anything to ensure the API
+ * itself is present. The most common tasks in configuration are:
+ *
+ * - Installing manager plugins and making them available to a host.
+ * - Configuring which manager a host should use.
+ *
+ *
+ * @section installing_manager_plugins Installing Manager plugins
+ *
+ * @note To aid problem solving, the plugin system logs debug messages
+ * when it searches for, and loads plugins.
+ *
+ * Presently, OpenAssetIO only supports Python based plugins.
+ * These can be installed in one of two ways.
+ *
+ * @subsection installing_manager_plugins_pip Using pip
+ *
+ * Most Python-based plugins should support what is known as
+ * <a href="https://packaging.python.org/en/latest/specifications/entry-points/" target="_blank">"entry point" based loading</a>.
+ * This means they can be installed using the package installer for
+ * Python, and will then be automatically detected.
+ *
+ * These plugins are usually published to
+ * <a href="https://pypi.org" target="_blank">PyPI</a>, and so are
+ * easy to install. Simply use the `pip` module of the Python
+ * installation you wish to make the plugin available to:
+ *
+ * @code{.sh}
+ * python -m pip install <manager-plugin-package>
+ * @endcode
+ *
+ * If the plugin is not published, you may be able to install it from a
+ * local checkout of its source code:
+ *
+ * @code{.sh}
+ * python -m pip install <path/to/src>
+ * @endcode
+ *
+ * @warning Many hosts use their own embedded Python interpreter. You
+ * may need to consult the relevant documentation to determine its
+ * location, version, `sys.path` etc. and adjust the invocation
+ * of `python` accordingly.
+ *
+ * This mechanism is easiest, but as `pip` installs packages into a
+ * specific python version, it can be harder to manage with multiple
+ * hosts if each has its own interpreter/version.
```

Comment:
[Question][Minor] Is this actually true you think? I have no experience with the matter, but it seems to me managing this via the manual workflow would be similarly difficult, albeit different problems. (I wonder about plugins built to support only python 3.11> for example, pip would help guard against this, but the manual workflow wouldn't.)

Reply:
Yeah different versions of the interpreter would need a different plugin installation.  But different applications, each with their own interpreter, but all using the same Python version, would be easier to configure with a system-wide env var, than by independently installing the plugin into all of them.

Reply:
If you have mimumum python version support for a python-only plugin (a common approach, as we do), then this is _much_ easier with the plugin path approach.

Eg, BAL is 3.7-3.10 compatible by design. This means it could be installed once and used with a singular point of consistent configuration, with any host from CY2020-CY2023, without needing to worry about how each particular host manages its `sys.path`.

Reply:
Version management can also be easier with a non `pip` approach, as with, e.g. Rez, you can have a single installation point for all available plugin versions and the appropriate version for a project selected at runtime (though this is true with `PYTHONPATH` or `OPENASSETIO_PLUGIN_PATH`, just not using `pip`). 

Maybe worth qualifying that it obviously requires supporting plugins?

We will also need the manual install approach for non-python plugins, and that section will be common, so its partially setting the stage for later in terms of doc structure.

Reply:
Ill try and rephrase this para to be less opinionated and more factual, then people can make their own mind up.

Reply:
Simplified [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/860/commits/3f224a8f12b49f018a5381170791b3904fd76a09)